### ISSUE TITLE
Question on the None-start states

### ISSUE THREAD

**Issue by hoangle96** [jmschrei/pomegranate#810]
I'm working with a discrete HMM model with the HMM creation code is as below. When I call `model.dense_transition_matrix()` to check the transition matrix, I see the None-start state listed there. Also, when inferring the hidden state from a sequence of observations by calling `model.predict()`, the None-start states never appear. After reading the documentation, it seems like this hidden state is always the start state. As such

1. Can I consider the transition probability from this None-start to other "actual" states the stationary distribution?
2. Can I force the model to start from one of the actual states? Is that recommended?

```
from sklearn.utils import check_random_state
import pomegranate as pom
import numpy as np

def create_independent_dist(feature, seed):
    unique_val = np.unique(feature)
    init_dict = {}
    random_state = check_random_state(seed)
    init_prob = random_state.rand(len(unique_val),1)
    init_prob = init_prob/init_prob.sum()
    
    for i in range(len(unique_val)):
        init_dict[i] = init_prob[i].item()
    return pom.DiscreteDistribution(init_dict)

def create_dist_for_states(n_states, feature_list, seed):
    distributions = []
    i = 0
    for _ in range(n_states):
        dist_list = []
        for f in feature_list:
            dist = create_independent_dist(f, i)
            i += 1
            dist_list.append(dist)
        distributions.append(pom.IndependentComponentsDistribution(dist_list))
    return distributions

def init_hmm(n_components, feature_list, seed):
    random_state_trans = check_random_state(seed**seed)
    transitions = random_state_trans.rand(n_components, n_components)
    transitions = transitions/transitions.sum()
    
    random_state_start = check_random_state(seed**2)
    starts = random_state_start.rand(n_components)
    starts = starts/starts.sum()
    distributions = create_dist_for_states(n_components, feature_list, seed)
    model = pom.HiddenMarkovModel.from_matrix(transitions, distributions, starts)
    
    return model
```

Here is what I found about the None-start state from the documentation

The sequence of labels can include hidden states! However, a consequence of this is that each sequence of labels must begin with the start state because that is where each sequence begins with being aligned to the model. For instance, for the sequence of observations `[1, 5, 6, 2]` the corresponding labels would be `['None-start', 'a', 'b', 'b', 'a']` because the default name of a model is None and the name of the start state is `{name}-start`. Likewise, you will need to add the end state label at the end of each sequence if you want an explicit end state, making the labels `['None-start', 'a', 'b', 'b', 'a', 'None-end']`.

Also, just for my understanding, do labels here refer to the hidden states of the HMM? Many thanks!

**Comment by jmschrei** [comment]
You are right that `None-start` is, by default, the start state. The reason that you don't see it in `model.predict` is that the default prediction algorithm is maximum-a-posteriori (MAP) decoding and not Viterbi decoding (though Viterbi decoding is an option). MAP decoding involves assigning to each observation the highest probably non-silent state (technically the highest probably state of any type, but because silent states by definition cannot produce observations the probability is always 0). In contrast, Viterbi decoding returns the entire path, starting at the start state and ending at the end state (if given). 

1. I'm not precisely sure what you mean when you say "stationary." In my mind a stationary distribution is one that does not change over time. In that sense, an entire HMM is stationary but conditioned on the current state. Most HMM formulations involve a vector of start probabilities, which is equivalent to the transitions out of the start state.

2. Yes, you can! If you pass `start= <your start state>` it should work. However, I haven't tested it since implementing it many years ago, so it should work but let me know if it doesn't. There's no problem with setting some other state to be the start state, but usually I just put an edge with 100% probability going from the None-start state to the one I want to really start. 

3. Yes, the labels are the names of the state objects.

**Comment by hoangle96** [comment]
Hi Jacob, 

Thanks for answering my questions! The stationary distribution I was think about is in fact the start probabilities. Sorry for the confusion.
